services/tradeoff_analytics/v1.js

```diff
@@ -55,10 +57,11 @@ TradeoffAnalytics.prototype.dilemmas = function(params, callback) {
     options: {
       method: 'POST',
       url: '/v1/dilemmas',
-      body: omit(params,['metadata_header']),
+      body: omit(params,['metadata_header','generate_visualization']),
       headers: {
         'x-watson-metadata' : params.metadata_header
       },
+      qs: params.generate_visualization === false ? {'generate_visualization': false} : undefined,
```

Comment:
the reason I did it like that is that I wanted to go with a more 'strict'
approach, and accept only booleans. Maybe I should have added some
assertion to check this.

2015-12-24 18:36 GMT+02:00 German Attanasio Ruiz <notifications@github.com>:

> In services/tradeoff_analytics/v1.js
> <https://github.com/watson-developer-cloud/node-sdk/pull/191#discussion_r48420472>
> :
>
> >        headers: {
> >          'x-watson-metadata' : params.metadata_header
> >        },
> > +      qs: params.generate_visualization === false ? {'generate_visualization': false} : undefined,
>
> Can we change
>
> qs: params.generate_visualization === false ? {'generate_visualization': false} : undefined,
>
> to be:
>
> qs: pick(params, ['generate_visualization'] ),
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/watson-developer-cloud/node-sdk/pull/191/files#r48420472>
> .
>